client/src/pages/RoomPage.tsx

```diff
@@ -53,17 +53,25 @@ function RoomPage({ name }: { name: string }): JSX.Element {
     networkRef.current = _network;
   };
 
-  const toggleMute = () => {
+  const toggleMute = useCallback(() => {
     updateSelfUserInfo({ ...selfUserInfoRef.current, mute: !selfUserInfoRef.current.mute });
-  };
+  }, [selfUserInfoRef.current.mute]);
+
+  const handleLeaveClicked = useCallback(() => {
```

Comment:
Can we name these the same as in the ButtonsBar? ie: onLeaveClicked and onSettingsClicked
I'll make a new issue to refactor so that toggleActive and toggleStatus are the same as well for consistency.
It'll also allow us to use a spread operator to pass in the functions like:
```
const buttonsBarProps = { onSettingsClicked, onMuteClicked, etc }
<ButtonsBar {...buttonsBarProps} />
```

Reply:
> Can we name these the same as in the ButtonsBar? ie: onLeaveClicked and onSettingsClicked
> I'll make a new issue to refactor so that toggleActive and toggleStatus are the same as well for consistency.
> It'll also allow us to use a spread operator to pass in the functions like:
> ```
> const buttonsBarProps = { onSettingsClicked, onMuteClicked, etc }
> <ButtonsBar {...buttonsBarProps} />
> ```

Nvm yeah I get what you mean by separating the "action" (handle) and the "when" (on)